src/components/implementation/Makefile.subsubdir

```diff
@@ -74,7 +74,7 @@ $(COMPONENT): $(OBJS)
 #	$(info |     [LD]   $@: Loading stubs and cos libraries $(if $(strip $(LIB_LIBRARIES)),including $(LIB_LIBRARIES)) into $<)
 	$(info |     [LD]   $(COMPNAME) ($@): Linking $(if $(strip $(LIB_LIBRARIES)),$(LIB_LIBRARIES)) $^)
 	@$(LD) $(LDFLAGS) -r -o $@.$(TMP_STR) $^ $(IF_SERV_STUBS) $(IF_CLIENT_STUBS) $(LIB_ASM_MANDITORY) $(LIB_FLAGS)
-	@$(MUSLCC) $@.$(TMP_STR) $(MUSLCFLAGS) -Wl,-melf_i386 -Xlinker -r -o $@.$(TMP_STR2)
+	@$(MUSLCC) $@.$(TMP_STR) $(MUSLCFLAGS) -Xlinker -melf_i386 -o $@.$(TMP_STR2)
```

Comment:
Why do we remove -r here?

Reply:
I'm also not convinced that `-melf_i386` is the same as `W1,-melf_i386`.  Are we losing fidelity here?

Reply:
@Others `-r` is already in `MUSLCFLAGS`, no need to repeat it.

@gparmer it is not just `-melf_i386`, it is `-Xlinker -melf_i386`. GCC docs defines `-Xlinker` and `-Wl,` as both "Pass option as an option to the linker". The difference that I can discern between the two is that `-Xlinker` can be used to provide system-specific linker options that GCC does not recognize while `-Wl,` is able to pass multiple options to the linker by separating them with commas. We don't need that extra functionality here so I can switch it back to using `-Wl,` however we use `-Xlinker` in Makefile.comp, so I chose to go that route for consistency.